java/client/src/main/java/glide/internal/protocol/CommandRequest.java

```diff
@@ -0,0 +1,125 @@
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Represents a command request with routing information for JNI execution. Handles serialization of
+ * commands with routing metadata for glide-core processing.
+ */
+public class CommandRequest {
+    private final String commandName;
+    private final List<String> arguments;
+    private final RouteInfo routeInfo;
+
+    private CommandRequest(String commandName, List<String> arguments, RouteInfo routeInfo) {
+        this.commandName = commandName;
+        this.arguments = arguments;
+        this.routeInfo = routeInfo;
+    }
+
+    public String getCommandName() {
+        return commandName;
+    }
+
+    public List<String> getArguments() {
+        return arguments;
+    }
+
+    public RouteInfo getRouteInfo() {
+        return routeInfo;
+    }
+
+    /** Create command with automatic routing (recommended for most cases) */
+    public static CommandRequest auto(String command, String... args) {
+        return new CommandRequest(command, Arrays.asList(args), RouteInfo.auto());
+    }
+
+    /** Create command for all nodes in cluster */
+    public static CommandRequest forAllNodes(String command, String... args) {
+        return new CommandRequest(command, Arrays.asList(args), RouteInfo.allNodes());
+    }
+
+    /** Create command for all primary nodes */
+    public static CommandRequest forAllPrimaries(String command, String... args) {
+        return new CommandRequest(command, Arrays.asList(args), RouteInfo.allPrimaries());
+    }
+
+    /** Create command for random node */
+    public static CommandRequest forRandomNode(String command, String... args) {
+        return new CommandRequest(command, Arrays.asList(args), RouteInfo.random());
+    }
+
+    /** Create command routed by slot key */
+    public static CommandRequest forSlotKey(
+            String command, String slotKey, boolean preferReplica, String... args) {
+        return new CommandRequest(
+                command, Arrays.asList(args), RouteInfo.bySlotKey(slotKey, preferReplica));
+    }
+
+    /** Create command routed by slot ID */
+    public static CommandRequest forSlotId(
+            String command, int slotId, boolean preferReplica, String... args) {
+        return new CommandRequest(
+                command, Arrays.asList(args), RouteInfo.bySlotId(slotId, preferReplica));
+    }
+
+    /** Create command routed by address */
+    public static CommandRequest forAddress(String command, String host, int port, String... args) {
+        return new CommandRequest(command, Arrays.asList(args), RouteInfo.byAddress(host, port));
+    }
+
+    /** Serialize to protobuf format for JNI transmission */
+    public byte[] toBytes() {
+        try {
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+
+            // Write command name length and data
+            byte[] cmdBytes = commandName.getBytes(StandardCharsets.UTF_8);
```

Comment:
Is it worth having a lookup table that caches command names as bytes? It seems likely that users will reuse the same commands in the same program.